client/me/purchases/purchases-list-in-dataviews/index.tsx

```diff
@@ -74,65 +79,87 @@ function MembershipSubscriptions( {
 	return <MembershipsDataViews memberships={ memberships } />;
 }
 
-function isDataLoading( {
-	isFetchingUserPurchases,
+const PurchasesListDataView: React.FC<
+	PurchasesListProps & PurchasesListConnectedProps & WithStoredPaymentMethodsProps
+> = ( {
 	hasLoadedUserPurchasesFromServer,
-}: {
-	hasLoadedUserPurchasesFromServer: boolean;
-	isFetchingUserPurchases: boolean;
-} ) {
-	if ( isFetchingUserPurchases && ! hasLoadedUserPurchasesFromServer ) {
-		return true;
-	}
-}
-
-function ConciergeBanner( {
+	isFetchingUserPurchases,
+	getManagePurchaseUrlFor,
+	purchases,
+	subscriptions,
+	sites,
 	nextAppointment,
-	availableSessions,
 	isUserBlocked,
-}: Pick<
-	PurchasesListConnectedProps,
-	'nextAppointment' | 'availableSessions' | 'isUserBlocked'
-> ) {
-	return (
-		<PurchaseListConciergeBanner
-			nextAppointment={ nextAppointment ?? undefined }
-			availableSessions={ availableSessions }
-			isUserBlocked={ isUserBlocked }
-		/>
-	);
-}
+	availableSessions,
+	userId,
+} ) => {
+	const translate = useTranslate();
+	const {
+		data: transferredOwnershipPurchases = [],
+		isLoading,
+		isSuccess: hasLoadedTransferredOwnershipPurchases,
+	} = useGetJetpackTransferredLicensePurchases( { userId: userId || undefined } );
+
+	const isDataLoading = useCallback( () => {
+		if (
+			( isFetchingUserPurchases && ! hasLoadedUserPurchasesFromServer ) ||
+			( isLoading && ! hasLoadedTransferredOwnershipPurchases )
+		) {
+			return true;
+		}
+
+		return false;
+	}, [
+		hasLoadedUserPurchasesFromServer,
+		isFetchingUserPurchases,
+		isLoading,
+		hasLoadedTransferredOwnershipPurchases,
+	] );
+
+	const allPurchasesLoaded =
+		hasLoadedUserPurchasesFromServer && hasLoadedTransferredOwnershipPurchases;
+
+	const allPurchases = useMemo( () => {
+		if ( allPurchasesLoaded ) {
+			return [ ...( purchases || [] ), ...transferredOwnershipPurchases ];
```

Comment:
This works but I'm curious: could we just alter the purchases endpoint to return these purchases as well (and perhaps flag them with a special property)? Maybe it's not as easy as I imagine but it seems like it would greatly simplify all this logic and if we ever need to know the user's purchases in a different place then we'd have it without having to jump through extra hoops. Is there a concern that those purchases need to only be accessed explicitly?